Add support for export via POST form and make RefWorks export use it.

EreMaijala opened the pull request:
I've done this to work around several issues with the newer RefWorks. It doesn't handle diacritics properly and fails to import records with certain characters (e.g. **Hawaiʻi** in https://www.finna.fi/Record/jykdok.1170822) when the import is done with a callback URL using https scheme. An additional benefit is that the export works even if VuFind isn't internet-accessible.

I've reworded the message a bit to make it more clear. I also blindly updated export-favorites.phtml although I couldn't find where the AJAX method exportFavoritesAjax would be called. Is it perhaps something left over from previous versions and could now be removed?

demiankatz commented:
@EreMaijala, it definitely seems like the existing export code needs a bit of cleanup.

Regarding exportFavoritesAjax, I've confirmed that it is not used or needed for any variation of export functionality. I have removed it from master and merged those changes here.

Regarding the language files -- it seems to me that in the current master, both export_redirect and export_refworks are totally unused. I'm inclined to strip both of those out of all of the language files to prevent future confusion. I see that you have repurposed export_refworks here, but maybe it would be better to create a new string (like export_send) to ensure that all of the translations are refreshed with appropriate wording. What do you think? If you agree to this strategy, I'll go ahead and remove the language strings from master, then merge the change into here....

EreMaijala commented:
@demiankatz  I agree that your proposed strategy would be much cleaner, so please go ahead.

demiankatz commented on module/VuFind/src/VuFind/Export.php:
```diff
@@ -385,4 +386,45 @@ public function getActiveFormats($context = 'record')
+     */
+    public function getBulkExportTargetWindow($format)
+    {
+        return !empty($this->exportConfig->$format->targetWindow)
```
Is this a new undocumented feature? Should this be noted in export.ini somewhere, or am I just missing something?

EreMaijala replied:
Yes, comment added.

demiankatz commented on module/VuFind/src/VuFind/Export.php:
```diff
@@ -385,4 +386,45 @@ public function getActiveFormats($context = 'record')
+     *
+     * @return string
+     */
+    public function getBulkExportPostField($format)
```
Do you think we need BulkExport in the method names here given that this is in the context of an Export class, and these settings apply whether it's a single record being exported or a bulk batch? (I realize that at least one existing method may already be badly named based on these standards... just wondering if we should avoid propagating a mistake further, if indeed the existing naming is a mistake).

EreMaijala replied:
I've cleaned up the method names. Thanks!

EreMaijala commented:
I believe I've addressed the review comments.

2.) I think POST should work with EZproxy too, but there may be a need to configure something on the EZproxy side. 

3.) Maybe the problem is the export data seems to be missing the RT tag that should always be the first tag of a record? The wording at http://www.refworks.com/refworks/help/Refworks.htm#RefWorks_Tagged_Format.htm indicates that it's a mandatory field.

demiankatz commented:
Good call on the RT issue, that was my problem -- I've put a generic fix in place at https://github.com/vufind-org/vufind/commit/7888a4abd1b0400cee7245651156801d75b8602c to prevent it from happening in the future. Everything else looks good too -- merging now.
